### PR TITLE
Produce a slimmer sign image

### PR DESCRIPTION
**Pull request by qbarrand**
Get the sign-fiel binary from the builder image.
Place binaries in a well-known location.

/cc @chr15p @ybettan

### CODE REVIEW

File: Dockerfile.signimage [hunk 1]
```diff
@@ -17,9 +13,11 @@ COPY vendor vendor
...
 
-FROM registry.ci.openshift.org/ocp/builder:rhel-8-golang-1.18-openshift-4.11
+RUN dnf install -y kernel-devel
+
+FROM registry.redhat.io/ubi8/ubi-micro:8.7
 
-COPY --from=builder /workspace/signimage /
-COPY --from=ksource /usr/src/kernels/*/scripts/sign-file /sign-file
+COPY --from=builder /workspace/signimage /usr/src/kernels/*/scripts/sign-file /usr/local/bin
+USER 65534:65534
```
[the remark this answers is not in the document]
Comment (qbarrand) [reply Dockerfile.signimage:21]: No it is not:
```
sh-4.4# cat /etc/passwd 
root:x:0:0:root:/root:/bin/bash
bin:x:1:1:bin:/bin:/sbin/nologin
daemon:x:2:2:daemon:/sbin:/sbin/nologin
adm:x:3:4:adm:/var/adm:/sbin/nologin
lp:x:4:7:lp:/var/spool/lpd:/sbin/nologin
sync:x:5:0:sync:/sbin:/bin/sync
shutdown:x:6:0:shutdown:/sbin:/sbin/shutdown
halt:x:7:0:halt:/sbin:/sbin/halt
mail:x:8:12:mail:/var/spool/mail:/sbin/nologin
operator:x:11:0:operator:/root:/sbin/nologin
games:x:12:100:games:/usr/games:/sbin/nologin
ftp:x:14:50:FTP User:/var/ftp:/sbin/nologin
nobody:x:65534:65534:Kernel Overflow User:/:/sbin/nologin
```
The default user is `root`, obviously we do not want to run as that.

[the remark this answers is not in the document]
Comment (ybettan) [reply Dockerfile.signimage:21]: Ack. Thanks.

[the remark this answers is not in the document]
Comment (qbarrand) [reply Dockerfile.signimage:21]: It's `MAX_UNSIGNED_SHORT`, the highest number you can code with a short integer.

File: Dockerfile.signimage [hunk 2]
```diff
@@ -17,9 +13,11 @@ COPY vendor vendor
 # Build
 RUN make signimage
 
-FROM registry.ci.openshift.org/ocp/builder:rhel-8-golang-1.18-openshift-4.11
+RUN dnf install -y kernel-devel
+
+FROM registry.redhat.io/ubi8/ubi-micro:8.7
```
[the remark this answers is not in the document]
Comment (qbarrand) [reply Dockerfile.signimage:18]: As discussed offline, there is no indication that it is: [CI docs](https://docs.ci.openshift.org/docs/architecture/images/)

File: cmd/signimage/signimage.go [hunk 3]
```diff
@@ -44,7 +45,7 @@ func canonicalisePath(path string) string {
 
 func signFile(filename string, publickey string, privatekey string) error {
 	logger.Info("running /sign-file", "algo", "sha256", "privatekey", privatekey, "publickey", publickey, "filename", filepath.Base(filename))
-	out, err := exec.Command("/sign-file", "sha256", privatekey, publickey, filename).Output()
+	out, err := exec.Command("sign-file", "sha256", privatekey, publickey, filename).Output()
```
[the remark this answers is not in the document]
Comment (qbarrand) [reply cmd/signimage/signimage.go]: Fixed.

**Pull request merged**